### DISCUSSION TITLE
Query properties across relationship on lifecycle method

### DISCUSSION THREAD

**Discussion by nabeelvalley** [strapi/strapi#8518] [Legacy Topics]
Good day, I'm trying to use `beforeFind` to search using a calculated value from `afterFind`, this is the kind of condition I'm trying to work under and would like any suggestions on how I can accomplish this

> I'm using lifecycle methods as I want this functionality from the admin panel

I have the following two models:

```js
TypeOuter = {
  OuterData: string,
  InnerData: TypeInner,
  InnerDataName: string // populated from afterFind from TypeInner
}

TypeInner = {
  Name: string
}
```

During my `afterFind` method I do something like this:

```js
afterFind: async (results, params, populate) => {
  if (results && results.length > 0)
    results.forEach(r => r.InnerDataName= r.InnerData.Name)
}
```

Now, from the admin panel, I would like to be able to search based on the InnerDataName, when submitting the search I have the following that's sent to `beforeFind`:

```js
  params: {
    _limit: '10',
    _sort: 'Name:ASC',
    InnerDataName: 'Something',
    _start: '0',
    _where: [ {} ]
  },
  populate: undefined
```

What's a possible way that I can go about searching by the InnerDataName and still get the data? I'm not too clear on the `beforeFind` usage at al

> Note that searching by this field doesn't work as it's only calculated during the `afterFind`, I can potentially move this to `beforeFind` as long as it retains the calculated field. The reason I've got the `InnerDataName` showing is because it's a separate relationship/object but for the purpose of the search page on the admin panel it's kind of necessary that the user can at least see the `InnerDataName`

Any advice/recommendations would be appreciated

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
